contracts/SimpleSettlement.sol

```diff
@@ -177,4 +179,35 @@ contract SimpleSettlement is FeeTaker {
             }
         }
     }
+
+    /**
+     * @dev Calculates fee amounts depending on whether the taker is in the whitelist and whether they have an _ACCESS_TOKEN.
+     * @param order The user's order.
+     * @param taker The taker address.
+     * @param takingAmount The amount of the asset being taken.
+     * @param extraData The extra data has the following format:
+     * ```
+     * 2 bytes — integrator fee percentage (in 1e5)
+     * 1 bytes - integrator rev share percentage (in 1e2)
+     * 2 bytes — resolver fee percentage (in 1e5)
+     * 32 bytes - estimated taking amount
+     * 1 byte - protocol surplus fee (in 1e2)
+     * ```
+     */
+    function _getFeeAmounts(IOrderMixin.Order calldata order, address taker, uint256 takingAmount, uint256 makingAmount, bytes calldata extraData) internal override virtual returns (uint256 integratorFeeAmount, uint256 protocolFeeAmount, bytes calldata tail) {
+        (integratorFeeAmount, protocolFeeAmount, tail) = super._getFeeAmounts(order, taker, takingAmount, makingAmount, extraData);
```

Comment:
`extraData` contains `whitelist` in `FeeTaker._parseFeeData(...)`, but it's not utilized here as stated in the natspec description and tests

Reply:
Fixed in e7b5dea